pyglider/utils.py

```diff
@@ -369,10 +371,17 @@ def oxygen_concentration_correction(data, ncvar):
         ref_sal = 0
     else:
         ref_sal = float(oxy_yaml['reference_salinity'])
+    # Find the closest temperature and salinity measurements to oxygen measurements with pandas timestamp matching
+    df_oxy = pd.DataFrame({"time": data.time, "oxygen_concentration": data.oxygen_concentration})
+    df_oxy = df_oxy[~np.isnan(df_oxy.oxygen_concentration)]
+    df_ctd = pd.DataFrame({"time": data.time, "salinity": data.salinity, "potential_temperature": data.potential_temperature})
+    df_ctd = df_ctd[~np.isnan(df_ctd.salinity)]
+    # Match the nearest timestamp within 5 seconds. Oxygen values with no temp/sal within 5 seconds will become nan
+    df_nearest = pd.merge_asof(df_oxy, df_ctd, on="time", direction="nearest", tolerance=datetime.timedelta(seconds=5))
```

Comment:
seems a little clunky to go back and forth from pandas like this.   Maybe worth the simplification, but is there no equivalent in xarray?

Reply:
Good idea. I found a serviceable equivalent with [da.reindex](https://github.com/pydata/xarray/issues/3218)

I'll submit a different PR using that method